feat(firestore-bigquery-export): add collection group query support

superphil0 opened the pull request:
This allows users to generate schemas for subcollections that appear across multiple parent documents by using collection group queries instead of regular collection queries.

This loosely follows the pattern of the import script by adding the same  flag --query-collection-group

I did test this on my own production DB, so works for my purposes

- Add --query-collection-group flag to gen-schema-view script
- Support both regular collection and collection group queries in Gemini schema generation
- Update sampleFirestoreDocuments function with isCollectionGroupQuery parameter
- Add interactive prompt for collection group query selection
- Update documentation with examples and explanations
- Add comprehensive test coverage for new functionality
- Maintain backward compatibility with existing functionality

Feature request here: https://github.com/firebase/extensions/issues/2516

superphil0 committed:
feat(firestore-bigquery-export): add collection group query support

- Add --is-collection-group-query flag to gen-schema-view script
- Support both regular collection and collection group queries in Gemini schema generation
- Update sampleFirestoreDocuments function with isCollectionGroupQuery parameter
- Add interactive prompt for collection group query selection
- Update documentation with examples and explanations
- Add comprehensive test coverage for new functionality
- Maintain backward compatibility with existing functionality

This allows users to generate schemas for subcollections that appear across
multiple parent documents by using collection group queries instead of
regular collection queries.

superphil0 committed:
refactor: change flag name to --query-collection-group

- Update command line flag from --is-collection-group-query to --query-collection-group
- Update all documentation and examples with new flag name
- Update tests to use new flag name
- Maintain same functionality with cleaner flag naming

This provides a more concise and intuitive flag name for collection group queries.

cabljac commented on package.json:
```diff
@@ -39,5 +39,8 @@
   "lint-staged": {
     "*.{js,md,yml,ts,json,yaml}": "prettier --write"
+  },
+  "dependencies": {
```
Is this intentional? This seems to be breaking CI.

cabljac reviewed (changes requested):
Can you remove the local dependency on the change tracker

superphil0 committed:
Remove Firebase extensions dependency from package.json

superphil0 committed:
Remove firestore-bigquery-change-tracker dependency

superphil0 commented:
> Can you remove the local dependency on the change tracker

done

superphil0 committed:
fix test for collection group query

superphil0 committed:
- Modify the prompt to clarify the option for using a collection group query.

- Update the message for the collection path to include both collection and collection group options.

superphil0 commented on firestore-bigquery-export/scripts/gen-schema-view/src/config/interactive.ts:
```diff
@@ -112,6 +112,14 @@ export const questions = [
   },
+  {
+    message:
+      "Do you want to use a collection group query instead of a regular collection query?",
```
ok done

superphil0 commented on firestore-bigquery-export/scripts/gen-schema-view/src/config/interactive.ts:
```diff
@@ -112,6 +112,14 @@ export const questions = [
     when: (answers) => answers.useGemini,
     default: "schema",
   },
+  {
```
ok done

superphil0 commented:
@cabljac anything missing here?

cabljac commented:
@superphil0  thanks for your patience! lgtm.

cabljac reviewed (approved):
lgtm

IzaakGough reviewed (commented):
Manually tested and lgtm
